dimod/reference/composites/clipcomposite.py

```diff
@@ -0,0 +1,154 @@
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# =============================================================================
+"""
+A composite that clips problem variables below and above threshold. if lower and upper bounds is not given
+it does nothing.
+
+"""
+try:
+    import collections.abc as abc
+except ImportError:
+    import collections as abc
+
+
+from dimod.binary_quadratic_model import BinaryQuadraticModel
+from dimod.core.composite import ComposedSampler
+from dimod.sampleset import SampleSet
+
+__all__ = 'ClipComposite',
+
+
+class ClipComposite(ComposedSampler):
+    """Composite to clip variables of a problem
+
+    Clips the variables of a bqm and modifies linear and quadratic terms
+    accordingly.
+
+    Args:
+       sampler (:obj:`dimod.Sampler`):
+            A dimod sampler
+
+    Examples:
+       This example uses :class:`.ClipComposite` to instantiate a
+       composed sampler that submits a simple Ising problem to a sampler.
+       The composed sampler scales linear, quadratic biases and offset as
+       indicated by options.
+
+       >>> h = {'a': -4.0, 'b': -4.0}
+       >>> J = {('a', 'b'): 3.2}
+       >>> sampler = dimod.ClipComposite(dimod.ExactSolver())
+       >>> response = sampler.sample_ising(h, J, lower_bound=-2.0, upper_bound=2.0)
+
+    """
+
+    def __init__(self, child_sampler):
+        self._children = [child_sampler]
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def parameters(self):
+        param = self.child.parameters.copy()
+        param.update({'lower_bound': [], 'upper_bound': []})
+        return param
+
+    @property
+    def properties(self):
+        return {'child_properties': self.child.properties.copy()}
+
+    def sample(self, bqm, lower_bound=None, upper_bound=None, **parameters):
+        """ Clip and sample from the provided binary quadratic model.
+
+        if lower_bound and upper_bound are given variables with value above upper_bound or below lower_bound are clipped.
+
+        Args:
+            bqm (:obj:`dimod.BinaryQuadraticModel`):
+                Binary quadratic model to be sampled from.
+
+            lower_bound (number):
+                Value by which to clip the variables from below.
+
+            upper_bound (number):
+                Value by which to clip the variables from above.
+            **parameters:
+                Parameters for the sampling method, specified by the child sampler.
+
+        Returns:
+            :obj:`dimod.SampleSet`
+
+        """
+        child = self.child
+        bqm_copy = _clip_bqm(bqm, lower_bound, upper_bound)
+        response = child.sample(bqm_copy, **parameters)
+
+        return SampleSet.from_samples_bqm(response.record.sample, bqm)
+
+    def sample_ising(self, h, J, offset=0.0, lower_bound=None, upper_bound=None, **parameters):
```

Comment:
This is handled by the abstract base class, think you can get rid of `sample_ising` alltogether.